components/dps310/include/dps310.h

```diff
@@ -0,0 +1,789 @@
+
+/**
+ * @brief Free the device descriptor.
+ *
+ * The device descriptor must NOT be NULL. `dps310_free_desc()` does not
+ * `free()` the device descriptor on error.
+ *
+ * @param[out] dev Device descriptor
+ * @return `ESP_OK` on success
+ */
+esp_err_t dps310_free_desc(dps310_t *dev);
+
+/**
+ * @brief Initialize DPS310 module
+ *
+ * The function does the followings:
+ *
+ * - read the DPS310_REG_ID, and identify the product ID. return ESP_FAIL if
+ *   the product ID does not match expected product ID.
+ * - reset the chip
+ * - perform a quirk
+ *
+ * @param dev[in] Device descriptor
+ * @param config[in] Configuration
+ * @return `ESP_OK` on success
+ */
+esp_err_t dps310_init(dps310_t *dev, dps310_config_t *config);
+
+/**
+ * @brief Reset the device.
+ *
+ * Perform "soft reset" and ensure the chip is fully functional by delaying
+ * `DPS310_STARTUP_DELAY_MS`.
+ *
+ * @param[in] dev Device descriptor
+ * @return `ESP_OK` on success, `ESP_ERR_INVALID_ARG` when `dev` and/or `config` is NULL, or other errors when I2C communication fails.
+ */
+esp_err_t dps310_reset(dps310_t *dev);
+
+/**
+ * @brief Get pressure measurement rate.
+ *
+ * @param[in] dev pointer to the device descriptor
+ * @param[out] value the value in the resister
+ * @return `ESP_OK` on success. `ESP_ERR_INVALID_ARG` when `dev` and/or
+ * `value` is NULL, or other errors when I2C communication fails.
+ */
+esp_err_t dps310_get_pm_rate(dps310_t *dev, uint8_t *value);
+
+/**
+ * @brief Set pressure measurement rate.
+ *
+ * @param[in] dev The pointer to the device descriptor.
+ * @param[in] value The value to set.
+ * @return `ESP_OK` on success, `ESP_ERR_INVALID_ARG` when `dev` and/or `config` is NULL, or other errors when I2C communication fails.
+ */
+esp_err_t dps310_set_pm_rate(dps310_t *dev, dps310_pm_rate_t value);
+
+/**
+ * @brief Get temperature measurement rate.
+ *
+ * @param[in] dev pointer to the device descriptor
+ * @param[out] value the value in the resister
+ * @return `ESP_OK` on success. `ESP_ERR_INVALID_ARG` when `dev` and/or
+ * `value` is NULL, or other errors when I2C communication fails.
+ */
+esp_err_t dps310_get_tmp_rate(dps310_t *dev, uint8_t *value);
+
+/**
+ * @brief Set temperature measurement rate.
+ *
+ * @param[in] dev The pointer to the device descriptor.
+ * @param[in] value The value to set.
+ * @return `ESP_OK` on success, `ESP_ERR_INVALID_ARG` when `dev` and/or `config` is NULL, or other errors when I2C communication fails.
+ */
+esp_err_t dps310_set_tmp_rate(dps310_t *dev, dps310_tmp_rate_t value);
+
+/**
+ * @brief Get pressure oversampling rate.
+ *
+ * @param[in] dev pointer to the device descriptor
+ * @param[out] value the value in the resister
+ * @return `ESP_OK` on success. `ESP_ERR_INVALID_ARG` when `dev` and/or
+ * `value` is NULL, or other errors when I2C communication fails.
+ */
+esp_err_t dps310_get_pm_prc(dps310_t *dev, uint8_t *value);
```

Comment:
i really don't like `pm_prc`, which is what the datasheet called. `p_oversampling` might be better.